### ISSUE TITLE
Consider indicator for SDK add-ons in v26-release

### ISSUE THREAD

**Issue by wolfbeast** [MoonchildProductions/Pale-Moon#592]
It may be a good idea to add the SDK warning in the add-on manager in v26 to prepare extension users for the discontinuation of SDK extensions in v27.
The warning text itself doesn't have to be localized, but v26 does have a different XPInstall addon storage method.

Maybe @wsdfhjxc can lend a hand -- the v27 code was theirs and it needs some work to port across.

**Comment by wsdfhjxc** [comment]
A database rebuild is needed to include additional column for a new property. In Tycho this is just a JSON file and if this property is missing then it gets a null value and nothing wrong happens. But this is quite risky because if the extensions.sqlite doesn't rebuild then the add-ons environment gets broken because of failed queries containing the new column which doesn't exist in previous database schema. 

The extensions' meta data gets updated during application update and the database should rebuild on DB_SCHEMA value change. I had some problems with the database but apparently managed to get this work (?). But I'm not sure how is this going to behave in a real update scenario and it's possible that it may get things worse. So I'll just leave the commit as a reference here for now.

**Comment by wolfbeast** [comment]
Yeah.. I don't particularly feel like having to do releng with a database migration path, because that would need to have a migration routine written.

Although slower, it may instead be possible to check all extensions on-the-fly from the extensions front-end/UI code and add warnings as-applicable there?

**Comment by wsdfhjxc** [comment]
The extensions.sqlite seems to be automatically rebuilt and updated with previous data when DB_SCHEMA value saved in user's profile doesn't match the value in modules shipped with the browser ([1](https://github.com/MoonchildProductions/Pale-Moon/blob/b8a683a5e659c3ede269bb77e0bca5f02d1ed3a0/toolkit/mozapps/extensions/XPIProvider.jsm#L3135-L3136), [2](
https://github.com/MoonchildProductions/Pale-Moon/blob/b8a683a5e659c3ede269bb77e0bca5f02d1ed3a0/toolkit/mozapps/extensions/XPIProviderUtils.js#L539-L605)). But it would need to be thoroughly tested. Although I ended up with a working thing, I had encountered some problems with this and running post-update compatibility check before, so now I'm not really sure about this.

With this solution the extension's .xpi must be opened to check if it contains Jetpack/SDK relatives. This is being done for existing extensions on their metadata update (forced to do it on application update) or for a new extension when it's installed. This is a one-time check and property assignment with a result stored in the database for future access. Doing this every time Add-ons Manager is opened would be probably slow and irritating, especially if someone has dozens of extensions. Also, I wouldn't know how to do it on-the-fly, because I just plugged into an existing method for parsing .xpi archives.

I'm sorry I couldn't help you much.
